solver/src/driver.rs

```diff
@@ -334,8 +337,13 @@ impl Driver {
             .as_u64();
         let liquidity = self
             .liquidity_collector
-            .get_liquidity(current_block_during_liquidity_fetch.into())
+            .get_liquidity(
+                current_block_during_liquidity_fetch.into(),
+                &self.inflight_orders,
+            )
             .await?;
+        // Reset the inflight orders after use.
```

Comment:
This way they won't be reset if e.g. get_liquidity fails and returns early above. Maybe this needs to be written to a temp object and reset first thing in the run_single method?

Reply:
I would actually argue this shouldn't get cleared at all. It already get overwritten here: https://github.com/gnosis/gp-v2-services/blob/e0abb426d00af2fc7c4ff70d41ffbe3974d153fe/solver/src/driver.rs#L424-L430